### ISSUE TITLE
TypeError: 'NoneType' object is not subscriptable

### ISSUE THREAD

**Issue by APOC00X** [littlethunder/digitsOCR#4]
TypeError: 'NoneType' object is not subscriptable

TypeError Traceback (most recent call last)
~/Desktop/.../ocrlearnKnn.py in ()
65 while True:
66 ret, frame = cap.read()
---> 67 frame = frame[:,:426]
68 rois, edges = findRoi(frame, 50)
69 digits = []
TypeError: 'NoneType' object is not subscriptable

run with jupyter notebook 5.50 in anaconda
python 3.6.5
cv　3.4.1
Macbook Air

**Comment by APOC00X** [comment]
TypeError                                 Traceback (most recent call last)
~/Desktop/.../ocrlearnKnn.py in <module>()
     65 while True:
     66     ret, frame = cap.read()
---> 67     frame = frame[:,:426]
     68     rois, edges = findRoi(frame, 50)
     69     digits = []

TypeError: 'NoneType' object is not subscriptable

**Comment by yiminzme** [comment]
At ocrlearnKnn.py:58, try other parameter such as 0, 2 or 3, etc. That is the parameter for camera index. Typically, parameter 0 would work for a computer with one camera.

cap = cv2.VideoCapture(0)

**Comment by APOC00X** [comment]
Thank you very much
That worked but, another error happened...

error                                     Traceback (most recent call last)
~/Desktop/.../digitsOCR-master/ocrlearnKnn.py in <module>()
     77     newFrame = np.hstack((frame,newEdges))
     78     cv2.imshow('frame', newFrame)
---> 79     videoFrame.write(newFrame)
     80     key = cv2.waitKey(1) & 0xff
     81     if key == ord(' '):

error: OpenCV(3.4.1) /opt/conda/conda-bld/opencv-suite_1527005509093/work/modules/videoio/src/cap_mjpeg_encoder.cpp:468: error: (-215) imgWidth == frameWidth && imgHeight == frameHeight && channels == 3 in function write

I tried to move those '#'s before the defining of 'width' and 'height' but no effect...

**Comment by yiminzme** [comment]
Yeah, the width and height are fixed, so it might not work. You can try my version https://github.com/vinc96/digitsOCR, which automatically fetch the width and height information and works on my Windows and Mac. If you have a chance to try it, please tell me if it works.
I also created a pull request for solving this issue.

**Comment by APOC00X** [comment]
！！！That worked smoothly ! I would very much appreciate receiving your help !
